tests/src/JIT/Regression/JitBlue/GitHub_23199/GitHub_23199.cs

```diff
@@ -0,0 +1,195 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+// The test revealed some problems of GCStress infrastructure on platforms with multi reg returns (arm64, amd64 Unix).
+// It required GCStress=0xc and GcStressOnDirectCalls=1 to hit issues. The issues were with saving GC pointers in the return registers.
+// The GC infra has to correctly mark registers with pointers as alive and must not report registers without pointers.
+
+namespace GitHub_23199
+{
+    public class Program
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static Object TestCrossgenedReturnWith2PointersStruct()
+        {
+            ProcessStartInfo pi = new ProcessStartInfo();
+            // pi.Environment calls crossgened HashtableEnumerator::get_Entry returning struct that we need.
+            Console.WriteLine(pi.Environment.Count);
+            return pi;
+        }
+
+        struct TwoPointers
+        {
+            public Object a;
+            public Object b;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static TwoPointers GetTwoPointersStruct()
+        {
+            var a = new TwoPointers();
+            a.a = new String("TestTwoPointers");
+            a.b = new string("Passed");
+            return a;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static Object TestTwoPointers()
+        {
+            var a = GetTwoPointersStruct(); // Report both.
+            Console.WriteLine(a.a + " " + a.b);
+            return a;
+        }
+
+        struct OnePointer
+        {
+            public Object a;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static TwoPointers GetOnePointer()
+        {
+            var a = new TwoPointers();
+            a.a = new String("TestOnePointer Passed");
+            return a;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static Object TestOnePointer()
+        {
+            var a = GetOnePointer();  // Report one.
+            Console.WriteLine(a.a);
+            return a;
+        }
+
+        struct FirstPointer
+        {
+            public Object a;
+            public int b;
```

Comment:
My impression is that Jit doesn't like `StructLayout(LayoutKind.Explicit)`, so setting this it can change its behavior significantly. I would like to keep them as is.

Reply:
Not suggesting that -- but you might use native sized ints for the non-pointer fields, so all fields are pointer sized.

Reply:
Can I write this in c#?

Reply:
For now, you have to do something like:
```C#
#if BIT64
using nint = System.Int64;
#else
using nint = System.Int32;
#endif
```

Reply:
Done, thank you.